src/Generic.Repository/Repository/BaseRepositoryAsync.cs

```diff
@@ -26,166 +27,154 @@ public class BaseRepositoryAsync<TValue, TFilter, TContext> :
+        {
+            var predicate = await filter.
+                CreateGenericFilter<TValue, TFilter>(CacheService, token).
+                ConfigureAwait(false);
+
+            await CreateQueryFiltered(predicate, notTracking, token).
+                ConfigureAwait(false);
+        }
+
+        #endregion INTERNALS METHODS
+
         #region QUERIES
 
         public async Task<IReadOnlyList<TValue>> FilterAllAsync(
             TFilter filter,
-            bool enableAsNotTracking,
+            bool notTracking,
             CancellationToken token)
         {
             ThrowErrorIf.IsNullValue(filter, nameof(filter), nameof(FilterAllAsync));
 
-            await CreateQueryFiltered(filter, enableAsNotTracking, token).
-                                        ConfigureAwait(false);
+            await CreateQueryFiltered(filter, notTracking, token).ConfigureAwait(false);
 
-            return await Query.ToListAsync().
-                            ConfigureAwait(false);
+            return await Query.ToListAsync().ConfigureAwait(false);
         }
 
         public async Task<IReadOnlyList<TReturn>> FilterAllAsync<TReturn>(
             TFilter filter,
-            bool enableAsNotTracking,
+            bool notTracking,
             Func<IEnumerable<object>, IEnumerable<TReturn>> mapper,
             CancellationToken token)
         {
             ThrowErrorIf.IsNullValue(filter, nameof(filter), nameof(FilterAllAsync));
             ThrowErrorIf.IsNullValue(mapper, nameof(mapper), nameof(FilterAllAsync));
 
-            await CreateQueryFiltered(filter, enableAsNotTracking, token).
-                                        ConfigureAwait(false);
+            await CreateQueryFiltered(filter, notTracking, token).ConfigureAwait(false);
 
-            var list = await CreateList(enableAsNotTracking);
+            var list = await CreateList(notTracking, token).ConfigureAwait(false);
 
             return mapper(list).ToList();
         }
 
         public async Task<IReadOnlyList<TReturn>> GetAllAsync<TReturn>(
-            bool enableAsNotTracking,
+            bool notTracking,
             Func<IEnumerable<object>, IEnumerable<TReturn>> mapper,
             CancellationToken token)
         {
             ThrowErrorIf.IsNullValue(mapper, nameof(mapper), nameof(FilterAllAsync));
 
-            var list = await CreateList(enableAsNotTracking);
+            var list = await CreateList(notTracking, token).ConfigureAwait(false);
 
             return mapper(list).ToList();
         }
 
         public async Task<IReadOnlyList<TReturn>> GetAllByAsync<TReturn>(
-            bool enableAsNotTracking,
+            bool notTracking,
             Expression<Func<TValue, bool>> predicate,
             Func<IEnumerable<object>, IEnumerable<TReturn>> mapper,
             CancellationToken token)
         {
-            CreateQueryFiltered(predicate, enableAsNotTracking);
+            await CreateQueryFiltered(predicate, notTracking, token);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/guilhermecaixeta/Generic.Service.DotNetCore2.2/pullRequest?prid=4961311)